src/Symfony/Component/DependencyInjection/Compiler/InlineServiceDefinitionsPass.php

```diff
@@ -69,7 +69,7 @@ private function inlineArguments(ContainerBuilder $container, array $arguments,
             if (is_array($argument)) {
                 $arguments[$k] = $this->inlineArguments($container, $argument);
             } elseif ($argument instanceof ArgumentInterface) {
-                $argument->setValues($this->inlineArguments($container, $argument->getValues()));
+                $this->inlineArguments($container, $argument->getValues());
```

Comment:
Is there any case where this could break? For example, if someone uses their own implementation of `ArgumentInterface` for whatever reason?